lib/chef/provider/aws_security_group.rb

```diff
@@ -284,11 +317,12 @@ def update_outbound_rules(sg, vpc)
   def update_rules(desired_rules, actual_rules_list, authorize: nil, revoke: nil)
     actual_rules = {}
     actual_rules_list.each do |rule|
+      rule = Hash[rule.each_pair.to_a]
```

Comment:
Struct has a `#to_h` method that might be a bit clearer for readers: http://ruby-doc.org/core-2.4.1/Struct.html#method-i-to_h

It appears that goes as far back as 2.0 (http://ruby-doc.org/core-2.0.0/Struct.html) so it should be OK to use since we no longer support older ruby versions.